libs/wingsdk/src/std/struct.ts

```diff
@@ -0,0 +1,45 @@
+import { Validator } from "jsonschema";
+import { T1 } from "./generics";
+import { Json } from "./json";
+import { Code, InflightClient } from "../core";
+
+/**
+ * Shared behavior for all structs
+ *
+ * @typeparam T1
+ */
+export class Struct {
+  /**
+   * @internal
+   */
+  public static _toInflightType(): Code {
+    return InflightClient.forType(__filename, this.name);
+  }
+
+  /**
+   * Converts a Json to a Struct
+   *
+   * This macro takes a Json object and the Struct definition file name then calls the validate method
+   *
+   * @macro ((j, s) => {return s._validate(j)})($args$, $self$)
+   */
+  public static fromJson(json: Json): T1 {
+    json;
+    throw new Error("Macro");
+  }
+
+  /** @internal */
+  public static _getValidator(deps?: { [key: string]: any }): Validator {
```

Comment:
```suggestion
  public static _getValidator(deps: { [key: string]: any } = {}): Validator {
```

This way you can avoid the `if`